package/MDAnalysis/core/selection.py

```diff
@@ -794,7 +794,73 @@ def _sel_with_icodes(self, vals, icodes):
         return mask
 
 
+class BoolSelection(Selection):
+    """Range selection for boolean values"""
+
+    def __init__(self, parser, tokens):
+        values = grab_not_keywords(tokens)
+        if not values:
+            values = [True]
+        val = values[0]
+        if isinstance(val, str):
+            val = val.lower()
+            if val == "false":
+                val = False
+            elif val == "true":
+                val = True
+        self.value = val
+
+    def apply(self, group):
+        vals = getattr(group, self.field)
+        return group[vals == self.value].unique
+
+
+class FloatRangeSelection(Selection):
+    """Range selection for float values"""
+
+    def __init__(self, parser, tokens):
+        values = grab_not_keywords(tokens)
+        if not values:
+            raise ValueError("Unexpected token: '{0}'".format(tokens[0]))
+
+        uppers = []  # upper limit on any range
+        lowers = []  # lower limit on any range
+
+        for val in values:
+            try:
+                lower = float(val)
+                upper = None
+            except ValueError:
+                # check if in appropriate format 'lower:upper' or 'lower-upper'
+                selrange = re.match(r"(\d+\.?\d*)[:-](\d+\.?\d*)", val)
```

Comment:
How does the parser deal with negative numbers, namely will it handle
* -1
* -1 - -0.5
* -1 - 1
* 1 - -1   (yes, that's a dumb range)

I think the regex is missing the possibility of negative numbers.

(The same probably applies to the integer selections.)

Reply:
Furthermore, don't we normally allow space around the `-` or `:` separators?

Reply:
> How does the parser deal with negative numbers, namely will it handle

It does now.

> Furthermore, don't we normally allow space around the - or : separators?

Apparently not because it required a bit of fiddling to properly parse things. I think values get split by whitespace before they're passed to the selection classes (to allow things like "resname A B C"). Relatedly, "to" was also not supported previously (and would still not be supported for things like Resid).

> 1 - -1 (yes, that's a dumb range)

This one now converts to something like "-1 to 1" (instead of the alternative modular meaning ">= 1 or <= -1"